src/main/java/javaslang/collection/TraversableOnce.java

```diff
@@ -591,7 +592,16 @@ default T last() {
         if (isEmpty()) {
             return None.instance();
         } else {
-            return maxBy((t1, t2) -> f.apply(t1).compareTo(f.apply(t2)));
+            T tm = null;
+            U um = null;
+            for (T t : Iterator.ofAll(this)) {
+                U u = f.apply(t);
+                if (um == null || u.compareTo(um) >= 0) {
+                    um = u;
+                    tm = t;
+                }
+            }
+            return new Some<>(tm);
```

Comment:
Then let's also save the null-check. And to be conform with the previous impl and maxBy(Comparator), we need to use `>` instead of `>=` (see internals of reduce and foldLeft... the zero is the previous value - but here we use the current value)

```java
final Iterator<T> iter = Iterator.ofAll(this);
T tm = iter.next();
U um = f.apply(tm);
while (iter.hasNext()) {
    final T t = iter.next();
    final U u = f.apply(t);
    if (u.compareTo(um) > 0) {
        um = u;
        tm = t;
    }
}
return new Some<>(tm);
```